include/seqan3/alignment/pairwise/execution/execution_handler_parallel.hpp

```diff
@@ -127,6 +128,33 @@ class execution_handler_parallel
         assert(status == contrib::queue_op_status::success);
     }
 
+    /*!\brief Takes underlying range of sequence pairs and invokes an alignment on each instance.
+     * \tparam algorithm_t              The type of the alignment algorithm.
+     * \tparam indexed_sequence_pairs_t The type of underlying sequence pairs annotated with an index;
+     *                                  must model seqan3::detail::indexed_sequence_pair_range.
+     * \tparam delegate_type            The type of the callable invoked on the std::invoke_result of `algorithm_t`.
+     *
+     * \param[in] algorithm              The alignment algorithm to invoke.
+     * \param[in] indexed_sequence_pairs The range of underlying annotated sequence pairs to be aligned.
+     * \param[in] delegate               A callable which will be invoked on each result of the computed alignments.
+     */
+    template <typename algorithm_t, indexed_sequence_pair_range indexed_sequence_pairs_t, typename delegate_type>
+    void execute(algorithm_t && algorithm,
```

Comment:
this is the algorithm that is supposed to be called by the execution handler. That can be basically anything that is invocable, so it was called fn_t for function before. Since it is only used in the context of executing alignment algorithms it is now refined to be a more specific in the name.

Reply:
Then maybe add the comment, that it must model callable ?

Reply:
And what is the delegate thing ?

Reply:
> Then maybe add the comment, that it must model callable ?

I will have another invocable concept for this after all the open PRs with these changes are mereged. Then I can make sure that this works. Is that ok?

Ah and a delegate is another callable that is called on the result of the alignment algorithm. The term is a standard term for something that should be called on the result of something without jumping out of the process (as opposed to coroutines). https://stackoverflow.com/questions/9568150/what-is-a-c-delegate

Reply:
> I will have another invocable concept for this after all the open PRs with these changes are mereged. Then I can make sure that this works. Is that ok?

Sure.

> Ah and a delegate is another callable that is called on the result of the alignment algorithm. The term is a standard term for something that should be called on the result of something without jumping out of the process (as opposed to coroutines). https://stackoverflow.com/questions/9568150/what-is-a-c-delegate

Ok, and what is the purpose of this here/what is the use case?

Reply:
We have two interfaces for dealing with alignment results. The first one results a alignment result range and the second calls a user defined delegate on the result. Accordingly, there are two executors. The two way which we already know and the one way which will not implement the buffering. So both of them reuse the execution handlers which just get a delegate to call the alignment result on. The two way just passes a lambda that moves the result into the correct buffer location.